### ISSUE TITLE
Update the Campaign Properties -> Token Properties Tab

### ISSUE THREAD

**Issue by cwisniew** [RPTools/maptool#1238]
**Is your feature request related to a problem? Please describe.**
The current Token Properties Tab in the Campaign properties is pretty primitive requiring a somewhat esoteric text format to define properties and where they are displayed.
![image](https://user-images.githubusercontent.com/73173/73714193-14d29900-4760-11ea-9452-349a0cf8be73.png)
![image](https://user-images.githubusercontent.com/73173/73714441-cd004180-4760-11ea-975d-4c5fbb5fcd88.png)

**Describe the solution you'd like**
This should be something that is easier to edit, something like a table with the property name, the display on stat sheet properties (either as set of check boxes or a drop down box), and default value. 

**Additional context**
This could be combined with an update of Sight and Light tabs for Campaign Properties Dialog (issues pending)

**Comment by JamzTheMan** [comment]
Can we use FX to replace these "dialogs"?

**Comment by cwisniew** [comment]
Yes  but campaign properties would have to be made thread safe. I am not near  a computer at the moment but I can check how big a task that would be in the morning

**Comment by cwisniew** [comment]
See #1247 for discussion about thread safety changes required if JavaFX is to be used for these dialogs

**Comment by Illydth** [comment]
As an additional "Quality of Life", can we get a check upon hitting "OK" that there aren't pending changes (at least) in the Token Properties tab?  I cannot begin to explain how many lost minutes to hours of work have gone into modifying properties and then hitting "OK" to save the changes only to realize that OK is actually Cancel along with cancel and that Update is really OK.  (yea, I know that was confusing, intentionally so).

At the end of the day the standard GUI Consensus is that there are 2 buttons:  OK and Cancel.  OK saves changes, Cancel Aborts Changes.  Interfaces do often have an "update" button, but that acts as a "Update Current Settings and Stay in Place" vs. OK which is "Update Settings and Close Window."  

It's not that clicking an "update" button is so immeasurably harder to manage, it's positioning that's the problem.  Every Modern GUI Interface we use now adays has a very common place to click to save work and exit, and that common location is a bit of a "social contract".  Our hands/muscle memory is tuned to complete changes and click "Ok" in the bottom of the window on the left of the two buttons.

With "Ok" Acting only like "Close Dialog" (not that a name change would fix the problem, it wouldn't), that somewhat breaks the social contract.  "OK" in the case where you haven't clicked "update" first, acts as if you'd pressed CANCEL...which is why I, and I have to assume many others, especially newer users, lose work in campaign properties so frequently.

At minimum, "Ok" should pop a dialog up that says "You have open unsaved work, do you wish to continue?" (a dialog that is very reasonable to be able to turn off in preferences for those who are smarter than I am and can handle clicking something other than ok).

**Comment by Azhrei** [comment]
The purpose of this issue is to completely replace the UI.  The update/ok/cancel would then go away.

But I feel you’re pain — It’s happened to me, too.  In fact, just switching from one property type to another requires hitting Update and, as you say, there’s no warning that switching will lose data.

**Comment by Tdue21** [comment]
@cwisniew Isn't this issue closed with 1.14?

**Comment by cwisniew** [comment]
> @cwisniew Isn't this issue closed with 1.14?

yep! Thanks
